src/app/clusters/ota-requestor/GenericOTARequestorDriver.h

```diff
@@ -71,17 +72,25 @@ class GenericOTARequestorDriver : public OTARequestorDriver
     bool GetNextProviderLocation(ProviderLocationType & providerLocation, bool & listExhausted) override;
 
 protected:
-    void StartDefaultProviderTimer();
-    void StopDefaultProviderTimer();
-    void DefaultProviderTimerHandler(System::Layer * systemLayer, void * appState);
+    void StartPeriodicQueryTimer();
```

Comment:
I'm starting to wonder if we should have just one timer. There's a lot of repetitive code with the Start/Stop functions. You can still keep the enum that differentiates between the two types as well as the the intervals and the handlers. For instance `StartPeriodicQueryTimer` and `StartWatchdogTimer` could become:

```
void GenericOTARequestorDriver::StartTimer(SelectedTimer timerType)
{
    switch (timerType)
    {
    case SelectedTimer::kPeriodicQueryTimer:
        timerInterval = mPeriodicQueryTimeInterval;
        timerHandler = PeriodicQueryTimerHandler;
        break;
    case SelectedTimer::kWatchdogTimer:
        timerInterval = mWatchdogTimeInterval;
        timerHandler = WatchdogTimerHandler;
        break;
    }

    ChipLogProgress(SoftwareUpdate, "Starting timer %d, timeout: %u seconds", timerType, (unsigned int) mWatchdogTimeInterval);
    ScheduleDelayedAction(
        System::Clock::Seconds32(timerInterval),
        [](System::Layer *, void * context) {
            (static_cast<GenericOTARequestorDriver *>(context))->timerHandler(nullptr, context);
        },
        this);
 }
```

Reply:
Per discussion, will not make this change for now.